subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/DefaultJavaPluginConvention.java

```diff
@@ -91,7 +93,17 @@ private File getReportsDir() {
 
     @Override
     public JavaVersion getSourceCompatibility() {
-        return srcCompat != null ? srcCompat : JavaVersion.current();
+        if (srcCompat != null) {
+            return srcCompat;
+        } else if (toolchainSpec != null && toolchainSpec.isConfigured()) {
+            return JavaVersion.toVersion(toolchainSpec.getLanguageVersion().get().toString());
+        } else {
+            return JavaVersion.current();
+        }
+    }
+
+    public JavaVersion getRawSourceCompatibility() {
```

Comment:
`@Nullable` ?

Reply:
I thought about it but the whole package has not been taken care of with regards to null APIs